### PR TITLE
Adding frequency aggregate

### PR DESCRIPTION
**Pull request by WireBaron**
This change adds a aggregate found to use the most
commonly occuring element for any column.

### CODE REVIEW

File: extension/src/datum_utils.rs [hunk 1]
```diff
@@ -198,55 +198,98 @@ flat_serialize_macro::flat_serialize! {
...
+            where
+                A: SeqAccess<'de>,
+            {
+                // borrowed from serialization/types.rs
+                unsafe fn get_struct<T>(tuple: pg_sys::HeapTuple) -> *mut T {
+                    //((char *) ((TUP)->t_data) + (TUP)->t_data->t_hoff)
+                    let t_data: *mut u8 = (*tuple).t_data.cast();
+                    let t_hoff = (*(*tuple).t_data).t_hoff;
+                    t_data.add(t_hoff as usize).cast()
+                }
```
Comment (JLockerman) [inline extension/src/datum_utils.rs:253]: I think we should only have one `fn get_struct()`, probably in `serialization`

File: extension/src/datum_utils.rs [hunk 2]
```diff
@@ -198,55 +198,98 @@ flat_serialize_macro::flat_serialize! {
...
+                    let tuple =
+                        pg_sys::SearchSysCache1(pg_sys::SysCacheIdentifier_TYPEOID as _, oid as _);
+                    if tuple.is_null() {
+                        pgx::error!("no type info for oid {}", oid);
+                    }
+        
+                    let type_tuple: pg_sys::Form_pg_type = get_struct(tuple);
+                    typmod = (*type_tuple).typmodin;
+                    pg_sys::ReleaseSysCache(tuple);
+                }
```
Comment (JLockerman) [inline extension/src/datum_utils.rs:277]: You shouldn't need typmod, IIRC it's only used for choosing between different output formats, is largely ignored, and `-1` is always a valid default.

File: extension/src/datum_utils.rs [hunk 3]
```diff
@@ -198,55 +198,98 @@ flat_serialize_macro::flat_serialize! {
...
+        }
+        let count = self.iter().count();
+        let mut seq = serializer.serialize_seq(Some(count + 1))?;
+        seq.serialize_element(&self.type_oid.0)?;
+        for element in self.iter() {
+            let chars = unsafe{pg_sys::OutputFunctionCall(&mut flinfo, element)};
+            let cstr = unsafe {
+                std::ffi::CStr::from_ptr(chars)
+            };
+            seq.serialize_element(cstr.to_str().unwrap())?;
```
Comment (JLockerman) [inline extension/src/datum_utils.rs:223]: We should add a
```
// TODO currently this always serializes inner data as text. When we start working on more-efficient network serialization, or we start using this in a transition state, we should use the binary format if we don't need human-readable output.
```

File: extension/src/datum_utils.rs [hunk 4]
```diff
@@ -0,0 +1,619 @@
...
+                        pgx::error!("no type info for oid {}", oid);
+                    }
+        
+                    let type_tuple: pg_sys::Form_pg_type = get_struct(tuple);
+                    typmod = (*type_tuple).typmodin;
+                    pg_sys::ReleaseSysCache(tuple);
+                }
+
+                let mut data = vec![];
+                while let Some(next) = seq.next_element::<String>()? {
```
Comment (JLockerman) [inline extension/src/datum_utils.rs:280]:
```suggestion
                while let Some(next) = seq.next_element::<&str>()? {
```
or 
```suggestion
                while let Some(next) = seq.next_element::<&'de str>()? {
```

File: extension/src/datum_utils.rs [hunk 5]
```diff
@@ -198,55 +198,98 @@ flat_serialize_macro::flat_serialize! {
...
+                    let next = std::ffi::CString::new(next).unwrap();  // TODO: error handling
```
Comment (JLockerman) [inline extension/src/datum_utils.rs:281]: annoying that pg forces an extra allocation here, but them's the breaks

File: extension/src/datum_utils.rs (hunk 3 above)
Comment (WireBaron) [reply extension/src/datum_utils.rs:227]: Looks good, taking verbatim

File: extension/src/datum_utils.rs (hunk 1 above)
Comment (WireBaron) [reply extension/src/datum_utils.rs]: Agreed, I'll make the serialization one crate public.

File: extension/src/datum_utils.rs (hunk 2 above)
Comment (WireBaron) [reply extension/src/datum_utils.rs]: Good to know, this was something of a pain to get.

File: extension/src/datum_utils.rs (hunk 1 above)
Comment (WireBaron) [reply extension/src/datum_utils.rs]: Actually this is no longer needed now that I'm not getting typmod, so I'm not bothering making get_struct available.

File: extension/src/datum_utils.rs (hunk 5 above)
Comment (WireBaron) [reply extension/src/datum_utils.rs:265]: Yeah, I was annoyed with this as well.

File: extension/src/datum_utils.rs (hunk 4 above)
Comment (WireBaron) [reply extension/src/datum_utils.rs]: Went with `&str`

**Pull request merged**